public/API/API_GetRecentGameAwards.php

```diff
@@ -0,0 +1,117 @@
+ *    string      AwardKind                 "mastered', "completed", "beaten-hardcore", or "beaten-softcore"
+ *    datetime    AwardDate                 an ISO8601 timestamp string for when the award was granted
+ *    int         GameID                    unique identifier of the game
+ *    string      GameTitle                 title of the game
+ *    int         ConsoleID                 unique identifier of the console associated to the game
+ *    string      ConsoleName               name of the console associated to the game
+ */
+
+use App\Community\Enums\AwardType;
+use App\Models\Game;
+use App\Models\PlayerBadge;
+use App\Models\System;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Validator;
+
+$input = Validator::validate(Arr::wrap(request()->query()), [
+    'd' => ['sometimes', 'date', 'before_or_equal:today'],
+    'o' => ['sometimes', 'integer', 'min:0', 'nullable'],
+    'k' => ['sometimes', 'string'],
+]);
+
+$offset = $input['o'] ?? 0;
+$targetDate = $input['d'] ?? null;
+$awardKindsCsv = $input['k'] ?? '';
+
+// Any given kind which isn't included in $validAwardKinds will just be ignored.
+$validAwardKinds = ['beaten-softcore', 'beaten-hardcore', 'completed', 'mastered'];
+$awardKinds = explode(',', $awardKindsCsv);
+$awardKinds = array_filter($awardKinds, fn ($kind) => in_array($kind, $validAwardKinds, true));
+
+// Construct the initial base query, which pulls beaten and mastery awards.
+$baseQuery = PlayerBadge::where(function ($query) {
+    $query->where('AwardType', AwardType::Mastery)
+        ->orWhere('AwardType', AwardType::GameBeaten);
+});
+
+// If the consumer is trying to filter by a start date, add that filtering to the query.
+if ($targetDate !== null) {
+    $baseQuery->whereDate('AwardDate', '<=', $targetDate);
+}
+
+// If the consumer is trying to filter by specific award kinds, add that filtering to the query.
+$kindMapping = [
+    'beaten-softcore' => ['AwardType' => AwardType::GameBeaten, 'AwardDataExtra' => false],
+    'beaten-hardcore' => ['AwardType' => AwardType::GameBeaten, 'AwardDataExtra' => true],
+    'completed' => ['AwardType' => AwardType::Mastery, 'AwardDataExtra' => false],
+    'mastered' => ['AwardType' => AwardType::Mastery, 'AwardDataExtra' => true],
+];
+if ($awardKinds !== null) {
+    $baseQuery->where(function ($query) use ($awardKinds, $kindMapping) {
+        foreach ($awardKinds as $awardKind) {
+            if (isset($kindMapping[$awardKind])) {
+                $mapping = $kindMapping[$awardKind];
+                $query->orWhere(function ($q) use ($mapping) {
+                    $q->where('AwardType', $mapping['AwardType'])
+                        ->where('AwardDataExtra', $mapping['AwardDataExtra']);
+                });
+            }
+        }
+    });
+}
+
+// Now, actually make the fetch, which includes both a limit and the offset.
+$fetchedGameAwards = (clone $baseQuery)->orderBy('AwardDate', 'desc')
+    ->skip($offset)
+    ->limit(25)
```

Comment:
This restriction should be documented in the header comments somewhere. Maybe the Count return value:
```
 *  int         Count                       number of game awards returned in the response (maximum: 25)
```

Would it be desirable to have a count parameter that defaults to 25 with some reasonable upper limit?

Reply:
> Would it be desirable to have a count parameter that defaults to 25 with some reasonable upper limit?

Good idea. I've added a `c` param which defaults to 25 and has a maximum of 100. If a user gives a value above the maximum, rather than throw an error, we do:

```php
$count = min($count, 100)
```

which forces it down to 100.

Due to this change, I've documented the restriction on the `c` param.